tests/README.md

```diff
@@ -0,0 +1,105 @@
+# Welcome to the agenix-rekey testing framework
+
+---
+
+# Usage
+Run `nix flake check` in this subfolder.
+To run as specific subtest run:
+```bash
+nix build .#check.<system>.<test-name> -L
+```
+where `test-name` is the name of the subfolder of `check` that contains your test.
+
+# Adding new tests 
+To add a test add a new flake and a file called `test.sh` in a subfolder of `./cases`.
+You don't need to lock your flake, it will be tested with the current agenix version on disk, as
+of running the test.
+
+## test.sh
+
+This file should contain your testing. It will be executed in a VM inside a folder containing your configured flake.
+You have access to these helpers on your $PATH:
+
+| command | description |
+|---|---|
+|`agenix`| the final agenix wrapper, used for rekeying, generating, etc |
+|`agenixActivate`| The agenix activation scripts, use these for running the agenix activation to test if decryption works correctly|
+
+You should use these wrappers from $PATH.
+
+If you need any other programs you should add them to the `runtimeInputs` of
+the `testScript` derivation in [this](./default.nix) file.
+
+## Your flake
+
+The flake to test has to of course use `agenix-rekey` in some way.
+Additionally these limitation apply:
+
+### Flat Input
+
+The flake to test has to have a completely flat input structure.
+This means that all transitive inputs have to be declared as a first level input and the followed.
+
+This is invalid:
+
+```nix
+inputs= {
+    agenix.url = "github:ryantm/agenix";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+}
+```
+
+Agenix also has as input `nixpkgs` but also `darwin`, `home-manager` and `system`.
```

Comment:
```suggestion
Because apart from `nixpkgs`, agenix internally has some other inputs that are not explicitly defined here. Specifically, the missing inputs are `darwin`, `home-manager` and `system`.
```